lta/nersc_verifier.py

```diff
@@ -0,0 +1,240 @@
+        """NerscVerifier has no additional status to contribute."""
+        return {}
+
+    def _expected_config(self) -> Dict[str, Optional[str]]:
+        """NerscVerifier provides our expected configuration dictionary."""
+        return EXPECTED_CONFIG
+
+    async def _do_work(self) -> None:
+        """Perform a work cycle for this component."""
+        self.logger.info("Starting work on Bundles.")
+        work_claimed = True
+        while work_claimed:
+            work_claimed = await self._do_work_claim()
+        self.logger.info("Ending work on Bundles.")
+
+    async def _do_work_claim(self) -> bool:
+        """Claim a bundle and perform work on it."""
+        # 1. Ask the LTA DB for the next Bundle to be verified
+        # configure a RestClient to talk to the LTA DB
+        lta_rc = RestClient(self.lta_rest_url,
+                            token=self.lta_rest_token,
+                            timeout=self.work_timeout_seconds,
+                            retries=self.work_retries)
+        self.logger.info("Asking the LTA DB for a Bundle to verify at NERSC with HPSS.")
+        pop_body = {
+            "claimant": f"{self.name}-{self.instance_uuid}"
+        }
+        response = await lta_rc.request('POST', '/Bundles/actions/pop?dest=NERSC&status=verifying', pop_body)
+        self.logger.info(f"LTA DB responded with: {response}")
+        bundle = response["bundle"]
+        if not bundle:
+            self.logger.info("LTA DB did not provide a Bundle to verify at NERSC with HPSS. Going on vacation.")
+            return False
+        # process the Bundle that we were given
+        if await self._verify_bundle_in_hpss(lta_rc, bundle):
+            await self._add_bundle_to_file_catalog(bundle)
+            await self._update_bundle_in_lta_db(lta_rc, bundle)
+        return True
+
+    async def _add_bundle_to_file_catalog(self, bundle: BundleType) -> bool:
+        """Add a FileCatalog entry for the bundle, then update existing records."""
+        # configure a RestClient to talk to the File Catalog
+        fc_rc = RestClient(self.file_catalog_rest_url,
+                           token=self.file_catalog_rest_token,
+                           timeout=self.work_timeout_seconds,
+                           retries=self.work_retries)
+        # determine the path where the bundle is stored on hpss
+        data_warehouse_path = bundle["path"]
+        basename = os.path.basename(bundle["bundle_path"])
+        stupid_python_path = os.path.sep.join([self.tape_bath_path, data_warehouse_path, basename])
+        hpss_path = os.path.normpath(stupid_python_path)
+        # create a File Catalog entry for the bundle itself
+        file_record = {
+            "uuid": bundle["uuid"],
+            "logical_name": hpss_path,
+            "checksum": bundle["checksum"],
+            "locations": [
+                {
+                    "site": "NERSC",
+                    "path": hpss_path,
```

Comment:
There should be an `archive` option set here.

Reply:
We do use `"archive": True` for files within bundle archives, and give the location a path like:
$BUNDLE_FILE:$FILENAME

The comments in the LTA Design Doc said: `$SITE:$BUNDLE_PATH:$FILENAME` but I thought it was silly to put $SITE in the `"path"` when the `"site"` field right next to it communicated that already (and is the field where FC customers would expect to find that metadata).

That said, if we use `"archive"` to indicate 'file is part of an archive file`, then I don't think we should use the tag on the archive file itself.